addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2573,6 +2573,49 @@ var SnippetsMenu = Widget.extend({
             }
         });
 
+        //avoid dropzone collisions
+
+        function adjustMargin($zone, marginType, overlap, minMargin, maxMargin) {
+            let currentMargin = parseFloat($zone.css(marginType));
+            let newMargin = Math.min(Math.max(currentMargin + overlap, minMargin), maxMargin);
+            $zone.css(marginType, newMargin + "px");
+        }
+
+        // Sort the dropzones by their vertical position (top value)
+        $zones.sort(function(a, b) {
+            let rectA = a.getBoundingClientRect();
+            let rectB = b.getBoundingClientRect();
+            return rectA.top - rectB.top;
+        });
+
+        for (let i = 0; i < $zones.length - 1; i++) {
+            let $zone1 = $($zones[i]);
+            let rect1 = $zone1[0].getBoundingClientRect();
+
+            for (let j = i + 1; j < $zones.length; j++) {
+                let $zone2 = $($zones[j]);
+                let rect2 = $zone2[0].getBoundingClientRect();
+
+                // Check if the zones are within 30px vertically
+                if (rect2.top - rect1.bottom > 30) {
```

Comment:
Is `30` the size of dropzones? Get the value dynamically instead of hardcoding it: if the design changes, nobody will think about adapting this code.